src/repository/repository.action.js

```diff
@@ -36,7 +36,12 @@ export const getRepository = url => {
     dispatch({ type: GET_REPOSITORY.PENDING });
 
     return v3
-      .getJson(url, accessToken)
+      .get(url, accessToken)
+      .then(response => {
+        return response.status === 200
+          ? response.json()
+          : response.json().then(err => Promise.reject(err));
```

Comment:
`response.json()` will never throw an error, no matter `response.status === 200` or not, right? So I think `Promise.reject()` is enough. We don't need to call `response.json()` again.

Reply:
@chinesedfan if we just only using `Promise.reject()`, we will not pass any value from that. It will just call `.catch()` but didn't pass any data you get from API response. In this case we will not get the "Not Found" messages from github API.

That's why first i need to get `json()` data from API, then i pass that data in the `Promise.reject(err)`.

Make both if true (equal 200) or false (not equal 200) will send same json formatted data.

Reply:
@dhamanutd Oh, hope I understand your idea this time. Do you mean `response.json().then((json) => response.status === 200 ? json : Promise.reject(json))`?

Reply:
@chinesedfan yes, that's what i'm intended to

Reply:
Good. Can you update it? Then I think there are no problems from my side. Thanks for your patient revisions!

Reply:
@chinesedfan done